projects/examples/src/components/datagrid/datagrid-link.example.component.ts

```diff
@@ -153,23 +154,22 @@ export class DatagridLinkExampleComponent<R extends Record> {
         },
     ];
 
-    actionDisplayConfig: DatagridActionDisplayConfig = {
-        contextual: {
-            styling: ActionStyling.INLINE,
-            buttonContents: TextIcon.TEXT,
-            position: DatagridContextualActionPosition.TOP,
-        },
-        staticActionStyling: ActionStyling.INLINE,
-    };
+    actionDisplayConfig: DatagridActionDisplayConfig = getDefaultDatagridActionDisplayConfig();
 
     selectionType = GridSelectionType.Single;
 
     changeActionLocation(): void {
         if (this.actionDisplayConfig.contextual.position === DatagridContextualActionPosition.TOP) {
-            this.actionDisplayConfig.contextual.position = DatagridContextualActionPosition.ROW;
+            this.actionDisplayConfig = {
```

Comment:
I think this is more clear like this and simpler if we change the interface:

```
this.actionDisplayConfig = getDefaultDatagridActionDisplayConfig();
if(this.actionDisplayConfig.contextual.position === DatagridContextualActionPosition.TOP) {
    this.actionDisplayConfig.contextual.position = DatagridContextualActionPosition.ROW;
    this.selectionType = GridSelectionType.None;
} else {
    this.selectionType = GridSelectionType.Single;
}
```

Reply:
As discussed on Slack, I need to reset the actionDisplayConfig object for me to update the display flags and action used in HTML template as shown below:
```
@Input() set actionDisplayConfig(config: ActionDisplayConfig) {
        this._actionDisplayConfig = getDefaultActionDisplayConfig(config || {});
        const buttonContents = this.actionDisplayConfig.contextual.buttonContents;
        this.shouldShowIcon = (TextIcon.ICON & buttonContents) === TextIcon.ICON;
        this.shouldShowText = (TextIcon.TEXT & buttonContents) === TextIcon.TEXT;
        this.shouldShowTooltip = buttonContents === TextIcon.ICON;
        this.updateActionListsAndDisplayFlags();
    }
```